### ISSUE TITLE
bug: ion-checkbox don't let interact with label

### ISSUE THREAD

**Issue by JEricaM** [ionic-team/ionic-framework#29304]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Adding a link inside the checkbox label should let the user open the link.  

### Expected Behavior

A link/click inside a label in checkbox should work correctly.

### Steps to Reproduce

Add a ion.checkbox like that

```
<ion-checkbox
                                        value={value}
                                        name='name'
                                        onIonChange={(e) => this.handleChangeCheck(e)}
                                        label-placement="end">
                                        <div class="ion-text-wrap">My text <a href="https://google.it"> click here</a></div>
                                    </ion-checkbox>
```
                                    
If I click on the link part of the label the href don't triggers, instead only works onIonChange.
                                  

### Code Reproduction URL

https://d6xspp.stackblitz.io

### Ionic Info

Ionic:

   Ionic CLI : 7.1.1

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v18.15.0
   npm    : 9.5.0
   OS     : macOS Unknown

**Comment by liamdebeasi** [comment]
Thanks for the report. This can be achieved by calling `stopPropagation` when the link is clicked. We've had this question asked a few times before, so I decided to add an example to our documentation on how to do this.

I've opened https://github.com/ionic-team/ionic-docs/pull/3584 to add this demo. Here is a preview of the docs: https://ionic-docs-git-checkbox-link-ionic1.vercel.app/docs/api/checkbox#links-inside-of-labels

I am going to close this since this is already achievable in Ionic, but let me know if you have questions.
